tests/Integration/Behaviour/Features/Context/Domain/Product/ProductImageFeatureContext.php

```diff
@@ -364,6 +391,57 @@ function (string $shopReference): int {
         }
     }
 
+    /**
+     * @When /^I try to get every image details for product "([^"]*)" in every shop:$/
+     */
+    public function iTryToGetEveryImageDetailsForProductInEveryShop(string $productReference): void
+    {
+        $this->shopProductImagesCollection = $this->getQueryBus()->handle(new GetShopProductImages($this->getSharedStorage()->get($productReference)));
+    }
+
+    /**
+     * @Then /^I should have the followings image details:/
+     */
+    public function iShouldHaveTheFollowingsImageDetailsForShop(TableNode $tableNode)
+    {
+        $dataRows = $tableNode->getColumnsHash();
+        $productImagesByShop = [];
+        foreach ($dataRows as $dataRow) {
+            $shopId = (int) $this->getSharedStorage()->get(trim($dataRow['shopReference']));
+            $productImagesByShop[$shopId][] = new ProductImageInShop(
+                (int) $this->getSharedStorage()->get(trim($dataRow['image reference'])),
+                (int) $dataRow['cover'] === 1
+            );
+        }
+
+        $expectedShopProductImagesArray = array_map(
+            function (int $shopId, array $productImages): ShopProductImages {
+                return new ShopProductImages($shopId, ProductImageCollection::from(...$productImages));
+            },
+            array_keys($productImagesByShop),
+            $productImagesByShop
+        );
```

Comment:
Optional suggestion but maybe it would be nice to use a `@Transform` to handle this? Here is an example https://github.com/PrestaShop/PrestaShop/blob/a36e19fd28c8184252782f7d613c9a7e32784588/tests/Integration/Behaviour/Features/Context/Domain/Product/SpecificPriceContext.php#L71

Reply:
No problem